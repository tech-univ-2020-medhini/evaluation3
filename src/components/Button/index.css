.Button{
    background-color: rgb(93, 201, 59);
    height: 2rem;
    width: 10em;
    color: white;
    font-size: 1.1rem;
    border-radius: 12px;
    margin: 1rem;
}