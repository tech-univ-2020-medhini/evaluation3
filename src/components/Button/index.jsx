import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Button = (props) => {
  const [count, setCount] = useState(0);
  const { text, click } = props;
  //   const countHandler = () => {
  //     click();
  //     setCount(count + 1);
  //   };

  return (
    <div className="Button">
      <button data-testid="test-btn" onClick={click} type="button">
        {text}
      </button>
    </div>
  );
};
Button.propTypes = {
  text: PropTypes.string.isRequired,
  click: PropTypes.func.isRequired,

};

export default Button;
