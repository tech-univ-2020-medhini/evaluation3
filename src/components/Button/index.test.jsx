import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from './index';


describe('the button component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Button>Test text</Button>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should call the onClick prop when the button is clicked', () => {
    const onClickMock = jest.fn();
    const { getByTestId } = render(<Button click={onClickMock} />);

    fireEvent.click(getByTestId('test-btn'));
    expect(onClickMock).toHaveBeenCalled();
  });
});
