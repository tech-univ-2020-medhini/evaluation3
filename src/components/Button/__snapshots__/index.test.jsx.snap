// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the button component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="Button"
  >
    <button
      data-testid="test-btn"
      type="button"
    />
  </div>
</DocumentFragment>
`;
